### PR TITLE
Update DG test cases, use cases, user stories, enhancement

### PR DESCRIPTION
**Pull request by KengHian**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -422,41 +468,61 @@ testers are expected to do more *exploratory* testing.
 
    1. Download the jar file and copy into an empty folder
 
-   1. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
+   2. Double-click the jar file Expected: Shows the GUI with a set of sample contacts. The window size may not be optimum.
```
Comment (Quasant) [inline docs/DeveloperGuide.md:471]: I thought optimum was a grammer mistake but turns out its valid english?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -422,41 +468,61 @@ testers are expected to do more *exploratory* testing.
...
 
-1. Saving window preferences
+2. Saving window preferences
```
Comment (Quasant) [inline docs/DeveloperGuide.md:473]: Gotta check the numbering, is it just a limitation of the UG markdown format

**Review by Quasant — APPROVED** [review]
ya LGTM overall

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (KengHian) [reply docs/DeveloperGuide.md:471]: Interesting

**Pull request merged**